nixos/modules/services/web-apps/archtika.nix

```diff
@@ -0,0 +1,309 @@
+      '';
+    };
+
+    services.postgresql = {
+      enable = true;
+      package = pkgs.postgresql_16;
+      ensureDatabases = [ cfg.databaseName ];
+      authentication = lib.mkOverride 51 ''
+        # IPv4 local connections:
+        host    all    all    127.0.0.1/32    trust
+        # IPv6 local connections:
+        host    all    all    ::1/128         trust
+        # Local socket connections:
+        local   all    all                    trust
+      '';
+      extraPlugins = [ pkgs.postgresql16Packages.pgjwt ];
+    };
+
+    systemd.services.postgresql.path = builtins.attrValues {
+      inherit (pkgs) gnutar gzip;
+    };
+
+    services.nginx = {
+      enable = true;
+      recommendedProxySettings = true;
+      recommendedTlsSettings = true;
+      recommendedZstdSettings = true;
+      recommendedOptimisation = true;
+
+      appendHttpConfig = ''
+        limit_req_zone $binary_remote_addr zone=requestLimit:10m rate=5r/s;
+        limit_req_status 429;
+        limit_req zone=requestLimit burst=20 nodelay;
+
+        add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;
+        add_header X-Frame-Options "SAMEORIGIN" always;
+        add_header X-Content-Type-Options "nosniff" always;
+        add_header Referrer-Policy "strict-origin-when-cross-origin" always;
+        add_header Permissions-Policy "accelerometer=(),autoplay=(),camera=(),cross-origin-isolated=(),display-capture=(),encrypted-media=(),fullscreen=(self),geolocation=(),gyroscope=(),keyboard-map=(),magnetometer=(),microphone=(),midi=(),payment=(),picture-in-picture=(self),publickey-credentials-get=(),screen-wake-lock=(),sync-xhr=(self),usb=(),xr-spatial-tracking=(),clipboard-read=(self),clipboard-write=(self),gamepad=(),hid=(),idle-detection=(),interest-cohort=(),serial=(),unload=()" always;
+
+        map $http_cookie $auth_header {
+          default "";
+          "~*session_token=([^;]+)" "Bearer $1";
+        }
+      '';
+
+      virtualHosts = {
+        "${cfg.domain}" = {
+          useACMEHost = cfg.domain;
+          forceSSL = true;
+          locations = {
+            "/" = {
+              proxyPass = "http://localhost:${toString cfg.webAppPort}";
+            };
+            "/previews/" = {
+              alias = "/var/www/archtika-websites/previews/";
+              index = "index.html";
+              tryFiles = "$uri $uri/ $uri.html =404";
+            };
+            "/api/rpc/export_articles_zip" = {
+              proxyPass = "http://localhost:${toString cfg.apiPort}/rpc/export_articles_zip";
+              extraConfig = ''
+                default_type application/json;
+                proxy_set_header Authorization $auth_header;
+              '';
+            };
+            "/api/" = {
+              proxyPass = "http://localhost:${toString cfg.apiPort}/";
```

Comment:
localhost can reference either 127.0.0.1 or ::1 and if not both are functioning you can hit the occasional 500 for no good reason